libra_toolbox/neutron_detection/activation_foils/naa.py

```diff
@@ -0,0 +1,733 @@
+    for i in range(len(peak_ergs)):
+        peak_ind =  np.argmin(np.abs((peak_ergs[i]) - xvals))
+        guess_parameters += [hist[peak_ind],
+                             peak_ergs[i],
+                             search_width/(3 * len(peak_ergs))]
+
+    # print(guess_parameters)
+                        
+    parameters, covariance = curve_fit(gauss, 
+                                       xvals[search_start:search_end], 
+                                       hist[search_start:search_end], 
+                                       p0=guess_parameters) 
+    # print(parameters)
+
+    areas = []
+    peak_starts = []
+    peak_ends = []
+    all_peak_params = []
+    peak_amplitudes = []
+    for i in range(len(peak_ergs)):
+        peak_amplitudes += [parameters[2 + 3*i]]
+        mean = parameters[2 + 3*i + 1]
+        sigma = np.abs(parameters[2 + 3*i + 2])
+        peak_start = np.argmin(np.abs((mean - 3*sigma) - xvals))
+        peak_end = np.argmin(np.abs((mean + 3*sigma) - xvals))
+
+        peak_starts += [peak_start]
+        peak_ends += [peak_end]
+
+        # Use unimodal gaussian to estimate counts from just one peak
+        peak_params = [parameters[0], parameters[1], parameters[2 + 3*i], mean, sigma]
+        all_peak_params += [peak_params]
+        gross_area = np.trapz(gauss(xvals[peak_start:peak_end], *peak_params), 
+                              x=xvals[peak_start:peak_end])
+
+        # Cut off trapezoidal area due to compton scattering and noise
+        trap_cutoff_area = np.trapz(parameters[0] + parameters[1] * xvals[peak_start:peak_end],
+                                    x=xvals[peak_start:peak_end])
+        area = gross_area - trap_cutoff_area
+        areas += [area]
+
+    if plot:
+        colors = ['red', 'green', 'magenta', 'brown', 'cyan']
+        fig, ax = plt.subplots(figsize=[5,4])
+        ax.stairs(hist, bins)
+        ax.plot(xvals, gauss(xvals, *parameters))
+
+        erg_str = ''
+
+        for i in range(len(peak_ergs)):
+            # plot dashed vertical lines at edges of peak
+            # ax.plot([xvals[peak_starts[i]]]*2, [0, np.max(hist)], '--', color=colors[i])
+            # ax.plot([xvals[peak_ends[i]]]*2, [0, np.max(hist)], '--', color=colors[i])
+
+            trap_cutoff_ys = parameters[0] + parameters[1] * xvals[peak_starts[i]:peak_ends[i]]
+            # print('Plot gauss params: {}'.format(all_peak_params[i]))
+            ax.fill_between(xvals[peak_starts[i]:peak_ends[i]], gauss(xvals[peak_starts[i]:peak_ends[i]], *all_peak_params[i]), 
+                        trap_cutoff_ys, color=colors[i], alpha=0.5)
+            erg_str += ' {:.0f},'.format(peak_ergs[i])
+
+        ax.set_ylabel('Counts', fontsize=14)
+        ax.set_ylim(-np.max(peak_amplitudes)*0.2, np.max(peak_amplitudes)*1.5)
+        ax.set_xlabel('Energy [keV]', fontsize=14)
+        plt.xticks(fontsize=12)
+        plt.yticks(fontsize=12)
+        ax.set_title('Peak:{} keV'.format(erg_str), fontsize=14)
```

Comment:
It's bad practice to have a plot function mixed with an analysis function. Please take it out as a separate function